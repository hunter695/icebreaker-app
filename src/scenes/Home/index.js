import React, { Component } from 'react'
import axios from 'axios'
import Switcher from 'components/Switcher'
import styled from 'styled-components'
import Button from 'components/Button'
import LineDisplayCard from './components/LineDisplayCard'

const Main = styled.main`
  display: flex;
  flex-direction: column;
  > button {
    width: 160px;
    padding: 16px 0;
    align-self: center;
    margin-top: 30px;
  }
`

const Controls = styled.div`
  display: flex;
  justify-content: space-around;
`

export default class Home extends Component {
  state = {
    type: 'icebreaker',
    wild: false,
    renderButton: true,
    data: null,
  }
  onClickLike = (val) => () => {
    const { collection, _id } = this.state.data
    axios.post('/like',
      { collection, id: _id, amount: val })
      .then(() => this.updateLikes(val))
    const date = new Date()
    localStorage.setItem('voteTime', (date.getTime() / 1000))
    this.setState({
      renderButton: false,
    })
  }

  getLine = (wild) => (
    axios.get(`/${this.state.type}?wild=${wild}`)
      .then((response) => response.data)
  )

  updateLikes = (val) => {
    const data = this.state.data

    if (val < 0) {
      data.dislikes += 1
    } else {
      data.likes += 1
    }
    this.setState({
      data,
    })
  }

  switchType = () => {
    const type = this.state.type === 'icebreaker' ? 'pickup' : 'icebreaker'
    this.setState({
      type,
    })
  }

  switchSafety = () => {
    const wild = !this.state.wild
    const data = this.data
    this.setState({
      wild, data,
    })
  }

  renderData = () => {
    const voteTime = parseFloat(localStorage.voteTime)
    const date = new Date()
    let renderButton = true
    if (voteTime && ((date.getTime() / 1000) - voteTime) < 30) {
      renderButton = false
    }
    const wild = this.state.wild
    const result = this.getLine(wild)
    result.then((data) => {
      this.setState({
        data, renderButton,
      })
    })
  }

  render() {
    return (
      <Main>
        <Controls>
          <Switcher
            name="homeTypePicker"
            choices={['icebreaker', 'pickup']}
            checked={this.state.type === 'pickup'}
            onChange={this.switchType}
          />
          <Switcher
            name="homeSafetyPicker"
            choices={['safe', 'wildcard']}
            checked={this.state.wild}
            onChange={this.switchSafety}
          />
        </Controls>
        <Button
          onClick={this.renderData}
        >
          Break the Ice!
        </Button>
        <LineDisplayCard renderButton={this.state.renderButton} isWild={this.state.wild} data={this.state.data} onClickLike={this.onClickLike} />
      </Main>
    )
  }
}
