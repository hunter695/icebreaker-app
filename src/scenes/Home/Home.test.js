import React from 'react'
import renderer from 'react-test-renderer'
import webdriver, { By } from 'selenium-webdriver'
import { path } from 'chromedriver'
import chrome from 'selenium-webdriver/chrome'
import Home from './'


const service = new chrome.ServiceBuilder(path).build()
chrome.setDefaultService(service)

const driver = new webdriver.Builder()
    .withCapabilities(webdriver.Capabilities.chrome())
    .build()

describe('<Home />', () => {
  it('renders initial UI', () => {
    const tree = renderer.create(<Home />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('doesn\'t have a LineDisplayCard until the "break the ice" button is clicked', async () => {
    driver.navigate().to('http://localhost:8080')
    expect(await driver.findElements(By.css('.line-display-card')).length === 0)
    driver.findElement(By.css('button')).click()
    expect(await driver.findElements(By.css('.line-display-card')).length === 1)
  })
})
