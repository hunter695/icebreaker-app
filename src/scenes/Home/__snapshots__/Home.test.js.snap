// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders initial UI 1`] = `
<main
  className="czfxPD"
>
  <div
    className="jlRHmo"
  >
    <div
      className="hwytNh"
    >
      <span
        className="iFFOKQ"
      >
        icebreaker
      </span>
      <div
        className="eCjdJE"
      >
        <input
          id="homeTypePicker"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="homeTypePicker"
        />
      </div>
      <span
        className="fUluhI"
      >
        pickup
      </span>
    </div>
    <div
      className="hwytNh"
    >
      <span
        className="iFFOKQ"
      >
        safe
      </span>
      <div
        className="eCjdJE"
      >
        <input
          id="homeSafetyPicker"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="homeSafetyPicker"
        />
      </div>
      <span
        className="fUluhI"
      >
        wildcard
      </span>
    </div>
  </div>
  <button
    className="bNfATh"
    onClick={[Function]}
  >
    Break the Ice!
  </button>
</main>
`;
